# Objetivo

O objetivo desta análise é identificar padrões que diferenciam transações fraudulentas de transações legítimas. A linguagem de programação utilizada foi `Python`, e a bibloteca utilizada para a visualização de dados foi `Pandas`, responsável por fornecer estruturas de dados essenciais para a análise, como o `dataframe`. A biblioteca para manipulação de números foi `Numpy` e `StandardScaler` para padronização e separação para teste e treino do modelo.

# Passo 1: Coleta de dados

Utilizando o `read.csv` do **Pandas**, os dados do **dataset** que contém informações de transações realizadas por cartões de crédito são carregados e o resultado é armazenado no **dataframe** `df`.

In [1]:
import pandas as pd

url ="https://storage.googleapis.com/download.tensorflow.org/data/creditcard.csv"

df = pd.read_csv(url)

# Visualização das 5 primeiras linhas do dataset
df.head()

,Time,V1,V2,V3,V4,V5,V6,V7,V8,V9,...,V21,V22,V23,V24,V25,V26,V27,V28,Amount,Class
0,0.0,-1.359807,-0.072781,2.536347,1.378155,-0.338321,0.462388,0.239599,0.098698,0.363787,...,-0.018307,0.277838,-0.110474,0.066928,0.128539,-0.189115,0.133558,-0.021053,149.62,0
1,0.0,1.191857,0.266151,0.166480,0.448154,0.060018,-0.082361,-0.078803,0.085102,-0.255425,...,-0.225775,-0.638672,0.101288,-0.339846,0.167170,0.125895,-0.008983,0.014724,2.69,0
2,1.0,-1.358354,-1.340163,1.773209,0.379780,-0.503198,1.800499,0.791461,0.247676,-1.514654,...,0.247998,0.771679,0.909412,-0.689281,-0.327642,-0.139097,-0.055353,-0.059752,378.66,0
3,1.0,-0.966272,-0.185226,1.792993,-0.863291,-0.010309,1.247203,0.237609,0.377436,-1.387024,...,-0.108300,0.005274,-0.190321,-1.175575,0.647376,-0.221929,0.062723,0.061458,123.50,0
4,2.0,-1.158233,0.877737,1.548718,0.403034,-0.407193,0.095921,0.592941,-0.270533,0.817739,...,-0.009431,0.798278,-0.137458,0.141267,-0.206010,0.502292,0.219422,0.215153,69.99,0



Para a maior compreesão dos dados, podemos utilizar o `df.columns`, que lista todas as colunas do **dataset**.

In [2]:
df.columns

Index(['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount',
       'Class'],
      dtype='object')

Outra função que pode ser útil é a `df.info`, que informa a quantidade de colunas, tipo de dados, quantidade de valores nulos por coluna e o consumo de memória do **dataset**

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284807 entries, 0 to 284806
Data columns (total 31 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   Time    284807 non-null  float64
 1   V1      284807 non-null  float64
 2   V2      284807 non-null  float64
 3   V3      284807 non-null  float64
 4   V4      284807 non-null  float64
 5   V5      284807 non-null  float64
 6   V6      284807 non-null  float64
 7   V7      284807 non-null  float64
 8   V8      284807 non-null  float64
 9   V9      284807 non-null  float64
 10  V10     284807 non-null  float64
 11  V11     284807 non-null  float64
 12  V12     284807 non-null  float64
 13  V13     284807 non-null  float64
 14  V14     284807 non-null  float64
 15  V15     284807 non-null  float64
 16  V16     284807 non-null  float64
 17  V17     284807 non-null  float64
 18  V18     284807 non-null  float64
 19  V19     284807 non-null  float64
 20  V20     284807 non-null  float64
 21  V21     28

O **dataset** contém várias colunas numéricas que representam caracteristicas das transações bancárias, além de uma coluna que indica se a transação é ou não fraudulenta.

Cada registro é uma operação feita por cartão de crédito. A coluna `Time` representa o momento da transação. Não se trata de uma data comum, mas uma váriavel númerica que indica o tempo desde o início até a coleta de dados.

A coluna `V1` até a `V28` são variáveis transformadas. Os dados originais foram processados utilizando técnicas de PCA para preservar a privacidade.

A penúltima coluna se chama `Amount` e representa o valor da transação. Essa variável é importante, porque as fraudes podem ter padrões específicos de valor.

A coluna `Class`, a última coluna da visualização, indica se a transação é ou não fraudulenta, sendo 0 indicando uma transação legítima, e 1 indicando uma fraude.

Para verificar como as classes estão distribuidas em números inteiros, podemos usar o método `value_counts`, que contabiliza quantas vezes cada valor aparece na coluna.

In [4]:
df['Class'].value_counts()

,count
Class,
0,284315
1,492


Para calcular a proporção em porcentagem dos valores da coluna `Class`, podemos usar o `normalize=Tru`e.

In [5]:
df['Class'].value_counts(normalize=True)

,proportion
Class,
0,0.998273
1,0.001727


Como podemos ver, as transações legítimas representam 99,8% e apenas 0.17% são fraudes. Isso pode significar que o dataset está desbalanceado que é quando uma classe aparece mais que a outra. Isso pode fazer com que o modelo aprenda errado. Exemplo: O modelo pode responder sempre que não é fraude. Ele vai acertar a maioria das vezes, mas não conseguirá fazer o mais importante, que é detectar uma fraude.